python/src/scipp/plot/tools.py

```diff
@@ -117,21 +117,56 @@ def vars_to_err(v):
     return v
 
 
-def check_log_limits(lims=None, vmin=None, vmax=None, scale=None):
-    """
-    Check if a limit is negative when a "log" norm is used.
-    If so, set the lower limits as 1.0e-3 * max_value.
-    Input can either be:
-      - 2 values `vmin` and `vmax` -> return vmin, vmax
-      - a list of length 2 `lims` -> return [vmin, vmax]
-    """
-    if lims is not None:
-        vmin = lims[0]
-        vmax = lims[1]
-    if vmin is not None and vmax is not None:
-        if scale == "log" and vmin <= 0:
-            vmin = 1.0e-03 * vmax
-    if lims is not None:
-        return np.array([vmin, vmax])
+def find_log_limits(x):
+    """
+    To find log scale limits, we histogram the data between 1.0-30
+    and 1.0e+30 and include only bins that are non-zero.
+    """
+    volume = np.product(x.shape)
+    pixel = sc.reshape(x, dims=['pixel'], shape=(volume, ))
+    pixel.variances = None  # no variances allowed in coord for histogram
+    weights = sc.Variable(dims=['pixel'], values=np.ones(volume))
+    hist = sc.histogram(sc.DataArray(data=weights, coords={'order': pixel}),
+                        bins=sc.Variable(dims=['order'],
+                                         values=np.geomspace(1e-30,
+                                                             1e30,
+                                                             num=61),
+                                         unit=x.unit))
+    # Find the first and the last non-zero bins
+    has_data = hist.data > 0.0 * sc.units.one
+    inds = np.where(has_data.values)
```

Comment:
https://numpy.org/doc/stable/reference/generated/numpy.where.html says one should use `nonzero` directly?